rfcs/text/0006_management_section_service.md

```diff
@@ -0,0 +1,341 @@
+
+```ts
+// my_plugin/public/plugin.ts
+
+export class MyPlugin {
+  setup(core, { management }) {
+    // Registering a new link to a new section
+    const mySection = management.sections.register({
+      id: 'my-section',
+      title: 'My Main Section', // display name
+      description: 'hello', // not used in current UI, but possibly in future
+      order: 10,
+      euiIconType: 'iconName',
+    });
+    mySection.registerLink({
+      id: 'my-link',
+      title: 'My Link', // display name
+      description: 'hello', // not used in current UI, but possibly in future
+      order: 20,
+      async mount(context, params) {
+        const { renderApp } = await import('./my-section');
+        return renderApp(context, params);
+      }
+    });
+
+    // Registering a new link to an existing section
+    const kibanaSection = management.sections.get('kibana');
+    kibanaSection.registerLink({ id: 'my-kibana-section-link', ... });
+  }
+
+  start(core, { management }) {
+    // access all registered sections, filtered based on capabilities
+    management.sections.getAvailable();
+    // automatically navigate to any section link by id
+    management.sections.navigateToSectionLink('my-kibana-section-link');
+  }
+}
+ 
+// my_plugin/public/my-section.tsx
+
+export function renderApp(context, { sectionBasePath, element }) {
+  ReactDOM.render(
+    // `sectionBasePath` would be `/app/management/my-section/my-link`
+    <MyApp basename={sectionBasePath} />,
+    element
+  );
+ 
+  // return value must be a function that unmounts (just like Core Application Service)
+  return () => ReactDOM.unmountComponentAtNode(element);
+}
+```
+
+We can also create a utility in `kibana_react` to make it easy for folks to `mount` a React app:
+```ts
+// src/plugins/kibana_react/public/mount_with_react.tsx
+import { KibanaContextProvider } from './context';
+
+export const mountWithReact = (
+  Component: React.ComponentType<{ basename: string }>,
+  context: AppMountContext,
+  params: ManagementSectionMountParams,
+) => {
+  ReactDOM.render(
+    (
+      <KibanaContextProvider services={{ ...context }}>
+        <Component basename={params.sectionBasePath} />
+      </KibanaContextProvider>
+    ),
+    params.element
+  );
+
+  return () => ReactDOM.unmountComponentAtNode(params.element);
+}
+
+// my_plugin/public/plugin.ts
+import { mountWithReact } from 'src/plugins/kibana_react/public';
+
+export class MyPlugin {
+  setup(core, { management }) {
+    const kibanaSection = management.sections.get('kibana');
+    kibanaSection.registerLink({
+      id: 'my-other-kibana-section-link',
+      ...,
+      async mount(context, params) {
+        const { MySection } = await import('./components/my-section');
+        return mountWithReact(MySection, context, params);
```

Comment:
Minor nit. I think we can clarify the way `mountWithReact` works if we retain the same pattern from the above example on line 105.

```js
async mount(context, params) {
  const { MySection } = await import('./components/my-section');
  const unmountCallback = mountWithReact(MySection, context, params);
  return () => unmountCallback();
}
```

Reply:
resolved